### PR TITLE
feat(smart-contract-orders): hide sc presign orders on start

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Continuation of https://github.com/cowprotocol/cowswap/pull/2462

Applying the same hiding logic to:
- Safe bundle flow for SWAP
- Regular SWAP presign
- Regular LIMIT presign

# To Test

## Safe SWAP bundle flow

1. Load app as Safe app, 
2. Configure a short expiration time (5min)
3. Pick sell token that needs approval
4. Go all the way until Safe confirmation modal is displayed
5. Cancel tx
* There should not be a pop up nor COW sound
* Order should not be pending in presign state
6. Open Explorer and look for you account
* There should be a pending presign tx
7. Wait for the order to expire on Swap page
* There should not be a expiration pop up nor sound
8. Back at the Swap, go all the way and this time place the tx/sign execute it
* There should be a pop up and sound
* Order should be pending in the corresponding state (presign or open if tx was already mined)

## Regular SWAP flow

1. Use any smart contract wallet
2. Configure a short expiration time (5min)
3. Go all the way until confirmation shows in the wallet
4. Cancel it
* There should not be a pop up nor COW sound
* Order should not be pending in presign state
5. Open Explorer and look for you account
* There should be a pending presign tx
6. Wait for the order to expire on Swap page
* There should not be a expiration pop up nor sound
7. Back at the Swap, go all the way and this time place the tx/sign execute it
* There should be a pop up and sound
* Order should be pending in the corresponding state (presign or open if tx was already mined)

## Regular LIMIT flow

1. Repeat steps from previous section but in the limit orders page
2. Results should be the same

---

EthFlow is unchanged
EOA flow should not change, but would be good to make sure it still behaves as before

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — COMMENTED** [review]
Hey @alfetopito , great, indeed!
The only one issue I have noticed that: 
1. When Swap and limit order expires in the signing state (not signed by a user)
2. Then NPS modal appears (case = order is expired). 

Can you please fix this? 

Also, I noticed that the state is not synced between the browser' windows. I opened #2537 issue for this. 

Also, according to the [conversation ](https://cowservices.slack.com/archives/C0361CDG8GP/p1684859235739219) signing orders are loaded from API in a different browser (expected behavior). 

Thanks!

**Comment by alfetopito** [comment]
> Hey @alfetopito , great, indeed! The only one issue I have noticed that:
> 
>     1. When Swap and limit order expires in the signing state (not signed by a user)
> 
>     2. Then NPS modal appears (case = order is expired).
> 
> 
> Can you please fix this?

Elena, since I just messed a lot with the code that triggers the NPS in https://github.com/cowprotocol/cowswap/pull/2552, I'll merge as is and address the issue based off of that PR.

**Comment by alfetopito** [comment]
Addressed NPS issue on https://github.com/cowprotocol/cowswap/pull/2554

File: src/modules/limitOrders/services/tradeFlow/index.ts [hunk 1]
```diff
@@ -50,33 +51,47 @@ export async function tradeFlow(
...
-    logTradeFlow('LIMIT ORDER FLOW', 'STEP 5: add pending order step')
+    logTradeFlow('LIMIT ORDER FLOW', 'STEP 4: add pending order step')
     addPendingOrderStep(
       {
         id: orderId,
         chainId: params.chainId,
         order: {
           ...order,
-          presignGnosisSafeTxHash: params.isGnosisSafeWallet && presignTx ? presignTx.hash : undefined,
+          isHidden: !params.allowsOffchainSigning,
```
Comment (anxolin) [inline src/modules/limitOrders/services/tradeFlow/index.ts:61]: ubernit: it wouldn't have harmed a comment. I guess here, you only how as visible posted orders that are signed. presign flow don't include the signature in them. 

Maybe also nice to hint that in step 6 they are unhidden as long as `presignTx` succeeds

File: src/modules/swap/services/swapFlow/index.ts [hunk 2]
```diff
@@ -28,24 +29,38 @@ export async function swapFlow(
...
-
-    logTradeFlow('SWAP FLOW', 'STEP 5: add pending order step')
     addPendingOrderStep(
       {
         id: orderId,
         chainId: input.context.chainId,
         order: {
           ...order,
-          presignGnosisSafeTxHash: input.flags.isGnosisSafeWallet && presignTx ? presignTx.hash : undefined,
+          isHidden: !input.flags.allowsOffchainSigning,
```
Comment (anxolin) [inline src/modules/swap/services/swapFlow/index.ts:38]: same comment

**Review by anxolin — COMMENTED** [review]
Very nice solution 👍 

I guess I could reproduce an edge case here by checking the implementation. But TBH, I'm fine to leave it like this.

1. Send the tx to sign in the wallet
2. Pretend to break the WC bridge --> maybe disconnect would do the trick? It could happen organically (the connection can be lost). 
3. Sign the TX in the Safe

The code will not have a chance to "unhide" the order.

**Pull request merged**